core/src/identity/rsa.rs

```diff
@@ -107,6 +107,24 @@ impl PublicKey {
     }
 }
 
+fn to_hex(bytes: &[u8]) -> String {
+    let mut hex = String::with_capacity(bytes.len() * 2);
+
+    for byte in bytes {
+        write!(hex, "{:02x}", byte).expect("Can't fail on writing to string");
+    }
+
+    hex
+}
+
+impl fmt::Debug for PublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("PublicKey")
+            .field("pkcs1", &to_hex(&self.0))
+            .finish()
+    }
```

Comment:
not sure if it would be much more readable, but the bytes could be formatted directly to the formatter and the string allocation and `to_hex` could go away with:

```
write!(f, "PublicKey {{ pcks1: ")?;
for byte in &self.0 {
    write!(f, "{:02x}", byte)?;
}
write!(f, " }}")
```